src/main/java/hal/command/Command.java

```diff
@@ -0,0 +1,77 @@
+package hal.command;
+
+import hal.exceptions.InputException;
+
+/**
+ * Enumeration of commands that can be executed in the HAL9000 application.
+ */
+public enum Command {
+    BYE {
+        @Override
+        public boolean isExit() {
+            return true;
+        }
+    },
+    DEADLINE,
+    DELETE,
+    EVENT,
+    FIND {
+        @Override
+        public boolean isIgnoredHistory() {
+            return true;
+        }
+    },
+    LIST {
+        @Override
+        public boolean isIgnoredHistory() {
+            return true;
+        }
+    },
+    MARK,
+    TODO,
+    UNDO {
+        @Override
+        public boolean isIgnoredHistory() {
+            return true;
+        }
+    },
+    UNMARK,
+    REDO {
+        @Override
+        public boolean isIgnoredHistory() {
+            return true;
+        }
+    };
+
+    public boolean isExit() {
+        return false;
+    }
+
+    public boolean isIgnoredHistory() {
+        return false;
+    }
+
+    /**
+     * Parses and returns the appropriate command based on user input.
+     *
+     * @param input The user's input string.
+     * @return The corresponding `Command` enum value.
+     * @throws InputException If there is an issue processing the input or if the command is unrecognized.
+     */
+    public static Command processCommand(String input) throws InputException {
```

Comment:
Perhaps this method can be named `parseCommand` instead? I feel like the word 'process' does not explain the method accurately enough. I noticed the same issue in several other places too, and I think that replacing 'process' with 'parse' might be more sensible.